### PR TITLE
Add Tutorial to Person

### PR DESCRIPTION
**Pull request by liviamil**

### CODE REVIEW

**Comment by liviamil** [comment]
Merged from tp master branch to resolve merge conflicts

File: src/test/java/seedu/address/model/person/TutorialTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        assertThrows(NullPointerException.class, () -> Tutorial.isValidTutorial(null));
+
+        // invalid tutorial group
+        assertFalse(Tutorial.isValidTutorial("")); // empty string
+        assertFalse(Tutorial.isValidTutorial("A")); // only one character
+
+        // valid tutorial group
+        assertTrue(Tutorial.isValidTutorial("T08")); // 3 digits
+    }
+}
```
Comment (turretDive) [inline src/test/java/seedu/address/model/person/TutorialTest.java:34]: This tests seems to have failed the java ci, there might have been parser issues?

File: src/test/java/seedu/address/model/person/TutorialTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Tutorial(null));
+    }
+
+    @Test
+    public void constructor_invalidTutorial_throwsIllegalArgumentException() {
+        String invalidTutorial = "A";
+        assertThrows(IllegalArgumentException.class, () -> new Tutorial(invalidTutorial));
+    }
+
```
Comment (turretDive) [inline src/test/java/seedu/address/model/person/TutorialTest.java:21]: This tests seems to have failed the java ci, there might have been parser issues?

**Review by turretDive — COMMENTED** [review]
Seems like 9 tests have failed, maybe can try going through the code logic again or see if the parser has any issues

File: src/test/java/seedu/address/model/person/TutorialTest.java (hunk 2 above)
Comment (liviamil) [reply src/test/java/seedu/address/model/person/TutorialTest.java:21]: Thanks for pointing that out, it was a Tutorial `VALIDATION_REGEX` issue, issue fixed.

**Comment by liviamil** [comment]
Created a new pull request

**Pull request closed without merging**